DoSomething/gladiator
# Messages settings

#### What's this PR do?
This PR adds a new section for Settings and within that section there is a sub section for Default Messages Settings. It allows editing the default messages for when additional Contests are created.

#### How should this be manually tested?
`composer dump-autoload` reset the DB, re-migrate, and re-seed. Check that there's a new **messages_settings** table. Then login and go to Settings -> Messages and try editing the default messages. Next create a new Contest and see if the the edits to the defaults that are populated into the form are what you had previously changed.

#### Any background context you want to provide?
I attempted to create a generalized Settings system, with the intent that it could become a separate DS PHP package that we could pull into multiple projects. However, the complexity of Messages, with their multipart fields (subject, body, pro_tip, etc) made that approach too convoluted. Scrapped it but kept the code for a potential future implementation. 

While some of the code in this PR isn't ideal and could very well be refactored a bit, I wanted to get something up that works given I lost a bunch of time with the original approach.

#### What are the relevant tickets?
Fixes #250

---
@DoSomething/gladiator
